csfieldguide/chapters/content/en/big-data/sections/analysis.md

```diff
@@ -0,0 +1,129 @@
+
+{image file-path="img/chapters/distributed-computing-diagram.png" caption="true" alt="Image shows four computers with arrows pointing to and from each of them, to illustrate that they are connected by a network."}
+
+Separate computers communicate with each other over a network in distributed systems.
+
+{image end}
+
+**Clustered Computing**: A computer cluster is a set of connected computers or servers, typically in the same location, that work together so that they can be viewed as a single system.
+
+{image file-path="img/chapters/cluster-computing-diagram.png" caption="true" alt="Five computers are connected in a circle with a border surrounding all five, to show that it is viewed as one system."}
+
+Multiple computers are connected in a local network and act as one entity in a computer cluster.
+
+{image end}
+
+Each of these approaches processes large amounts of data in a much shorter amount of time than a single computer. They also require any problem they are working on to be broken down, or decomposed into sets of smaller problems or processes, which can each be tackled independently and simultaneously. Computer Scientists and Software Engineers need to come up with good methods to decompose their analysis processes, so that they can be effectively tackled with these approaches. Some problems are easy to decompose into independent pieces and are called *embarrassingly parallel*.
+
+An example is applying the same image smoothing filter to millions of pixels in an image, each computer or processor can apply the filter to a small section of the image and then the results can be recombined to form the new smoothed image. Other problems, such as weather prediction simulations, are harder to decompose, and require advanced techniques including pipe-lining, stochastic sampling, and replication.
+
+**MapReduce**
+
+MapReduce is an example of a general algorithmic method that makes the most of Parallel, Distributed, and Clustered Computing. MapReduce is named as such because it is made up of two steps: "map" and "reduce", and is a type of split-apply-combine strategy for data analysis.
+
+The best way to illustrate how a MapReduce algorithm works is to imagine you and five friends are making a sandwich. In the sandwich you want cucumber, tomato, lettuce, mayonnaise and of course bread on either end.
+The first stage is "Map". In this stage, you each take one of the fillings and prepare it to be put in the sandwich (slice the bread in half, wash the lettuce, cut the tomato, etc). This is much quicker than just one of you preparing each ingredient one at a time.
+Now that all the ingredients are prepared, you move on to the "Reduce" stage. This is often described as the "summary operation" because each of the 5 ingredients now need to be combined into one.
+And now we have a delicious MapReduce sandwich ready to eat.
```

Comment:
To be honest I wasn't really a fan of this example in general! :P it seems to be a common way people explain it though. 

I was actually wondering about taking out the MapReduce section entirely since it is really specific and I'm not really sure if it adds anything to the chapter. What do you think? I'll ask Tim as well.

Reply:
I think it could be left out

Also to pick at it further the 'slice the bread in half' thing seems kinda odd.  Cutting the bread in half *before* stuff is put in the sandwich would either end up with two half sandwiches connected by the ingredients that haven't been sliced in half yet, or this:

![image](https://user-images.githubusercontent.com/25916475/71036218-1a5d2b00-2182-11ea-8d23-a49f7d6ae73a.png)